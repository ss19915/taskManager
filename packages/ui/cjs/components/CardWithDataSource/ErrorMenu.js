"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _constants = require("./constants");

var _theme = require("@react-web-app/theme");

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var ErrorMenu =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(ErrorMenu, _React$PureComponent);

  function ErrorMenu() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, ErrorMenu);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(ErrorMenu)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      showError: false
    });

    _defineProperty(_assertThisInitialized(_this), "toggleShowError", function () {
      return _this.setState(function (prevState) {
        return {
          showError: !prevState.showError
        };
      });
    });

    return _this;
  }

  _createClass(ErrorMenu, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          retry = _this$props.retry,
          status = _this$props.status,
          error = _this$props.error;
      return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_theme.Button, {
        onClick: retry,
        disabled: status === _constants.Status.LOADING
      }, _constants.ErrorMenuLabels.RETRY), _react["default"].createElement(_theme.Button, {
        onClick: this.toggleShowError
      }, this.state.showError ? _constants.ErrorMenuLabels.HIDE_ERROR : _constants.ErrorMenuLabels.SHOW_ERROR), this.state.showError && _react["default"].createElement(_theme.Paper, {
        css: function css(theme) {
          return {
            color: theme.warningColor
          };
        }
      }, "Error: ".concat(JSON.stringify(error))));
    }
  }]);

  return ErrorMenu;
}(_react["default"].PureComponent);

_defineProperty(ErrorMenu, "propTypes", {
  retry: _propTypes["default"].func,
  status: _propTypes["default"].oneOf(Object.values(_constants.Status)),
  error: _propTypes["default"].object
});

var _default = ErrorMenu;
exports["default"] = _default;