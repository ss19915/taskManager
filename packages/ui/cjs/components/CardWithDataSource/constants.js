"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ErrorMenuLabels = exports.Status = void 0;
var Status = {
  LOADING: 0,
  LOADED: 1,
  ERROR: -1
};
exports.Status = Status;
var ErrorMenuLabels = {
  RETRY: 'Retry',
  HIDE_ERROR: 'Hide Error',
  SHOW_ERROR: 'Show Error'
};
exports.ErrorMenuLabels = ErrorMenuLabels;