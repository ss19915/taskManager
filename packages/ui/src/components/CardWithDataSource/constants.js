export const Status = {
    LOADING: 0,
    LOADED: 1,
    ERROR: -1,
};

export const ErrorMenuLabels = {
    RETRY: 'Retry',
    HIDE_ERROR: 'Hide Error',
    SHOW_ERROR: 'Show Error'
};