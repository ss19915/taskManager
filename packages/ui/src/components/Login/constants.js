export const COUNRTY_CODES = [{"name":"AD","code":"+376"},{"name":"AE","code":"+971"},{"name":"AF","code":"+93"},{"name":"AG","code":"+1268"},{"name":"AI","code":"+1 264"},{"name":"AL","code":"+355"},{"name":"AM","code":"+374"},{"name":"AN","code":"+599"},{"name":"AO","code":"+244"},{"name":"AQ","code":null},{"name":"AR","code":"+54"},{"name":"AS","code":"+1 684"},{"name":"AT","code":"+43"},{"name":"AU","code":"+61"},{"name":"AW","code":"+297"},{"name":"AX","code":""},{"name":"AZ","code":"+994"},{"name":"BA","code":"+387"},{"name":"BB","code":"+1 246"},{"name":"BD","code":"+880"},{"name":"BE","code":"+32"},{"name":"BF","code":"+226"},{"name":"BG","code":"+359"},{"name":"BH","code":"+973"},{"name":"BI","code":"+257"},{"name":"BJ","code":"+229"},{"name":"BL","code":"+590"},{"name":"BM","code":"+1 441"},{"name":"BN","code":"+673"},{"name":"BO","code":"+591"},{"name":"BR","code":"+55"},{"name":"BS","code":"+1 242"},{"name":"BT","code":"+975"},{"name":"BW","code":"+267"},{"name":"BY","code":"+375"},{"name":"BZ","code":"+501"},{"name":"CA","code":"+1"},{"name":"CC","code":"+61"},{"name":"CD","code":"+243"},{"name":"CF","code":"+236"},{"name":"CG","code":"+242"},{"name":"CH","code":"+41"},{"name":"CI","code":"+225"},{"name":"CK","code":"+682"},{"name":"CL","code":"+56"},{"name":"CM","code":"+237"},{"name":"CN","code":"+86"},{"name":"CO","code":"+57"},{"name":"CR","code":"+506"},{"name":"CU","code":"+53"},{"name":"CV","code":"+238"},{"name":"CX","code":"+61"},{"name":"CY","code":"+537"},{"name":"CZ","code":"+420"},{"name":"DE","code":"+49"},{"name":"DJ","code":"+253"},{"name":"DK","code":"+45"},{"name":"DM","code":"+1 767"},{"name":"DO","code":"+1 849"},{"name":"DZ","code":"+213"},{"name":"EC","code":"+593"},{"name":"EE","code":"+372"},{"name":"EG","code":"+20"},{"name":"ER","code":"+291"},{"name":"ES","code":"+34"},{"name":"ET","code":"+251"},{"name":"FI","code":"+358"},{"name":"FJ","code":"+679"},{"name":"FK","code":"+500"},{"name":"FM","code":"+691"},{"name":"FO","code":"+298"},{"name":"FR","code":"+33"},{"name":"GA","code":"+241"},{"name":"GB","code":"+44"},{"name":"GD","code":"+1 473"},{"name":"GE","code":"+995"},{"name":"GF","code":"+594"},{"name":"GG","code":"+44"},{"name":"GH","code":"+233"},{"name":"GI","code":"+350"},{"name":"GL","code":"+299"},{"name":"GM","code":"+220"},{"name":"GN","code":"+224"},{"name":"GP","code":"+590"},{"name":"GQ","code":"+240"},{"name":"GR","code":"+30"},{"name":"GS","code":"+500"},{"name":"GT","code":"+502"},{"name":"GU","code":"+1 671"},{"name":"GW","code":"+245"},{"name":"GY","code":"+595"},{"name":"HK","code":"+852"},{"name":"HN","code":"+504"},{"name":"HR","code":"+385"},{"name":"HT","code":"+509"},{"name":"HU","code":"+36"},{"name":"ID","code":"+62"},{"name":"IE","code":"+353"},{"name":"IL","code":"+972"},{"name":"IM","code":"+44"},{"name":"IN","code":"+91"},{"name":"IO","code":"+246"},{"name":"IQ","code":"+964"},{"name":"IR","code":"+98"},{"name":"IS","code":"+354"},{"name":"IT","code":"+39"},{"name":"JE","code":"+44"},{"name":"JM","code":"+1 876"},{"name":"JO","code":"+962"},{"name":"JP","code":"+81"},{"name":"KE","code":"+254"},{"name":"KG","code":"+996"},{"name":"KH","code":"+855"},{"name":"KI","code":"+686"},{"name":"KM","code":"+269"},{"name":"KN","code":"+1 869"},{"name":"KP","code":"+850"},{"name":"KR","code":"+82"},{"name":"KW","code":"+965"},{"name":"KY","code":"+ 345"},{"name":"KZ","code":"+7 7"},{"name":"LA","code":"+856"},{"name":"LB","code":"+961"},{"name":"LC","code":"+1 758"},{"name":"LI","code":"+423"},{"name":"LK","code":"+94"},{"name":"LR","code":"+231"},{"name":"LS","code":"+266"},{"name":"LT","code":"+370"},{"name":"LU","code":"+352"},{"name":"LV","code":"+371"},{"name":"LY","code":"+218"},{"name":"MA","code":"+212"},{"name":"MC","code":"+377"},{"name":"MD","code":"+373"},{"name":"ME","code":"+382"},{"name":"MF","code":"+590"},{"name":"MG","code":"+261"},{"name":"MH","code":"+692"},{"name":"MK","code":"+389"},{"name":"ML","code":"+223"},{"name":"MM","code":"+95"},{"name":"MN","code":"+976"},{"name":"MO","code":"+853"},{"name":"MP","code":"+1 670"},{"name":"MQ","code":"+596"},{"name":"MR","code":"+222"},{"name":"MS","code":"+1664"},{"name":"MT","code":"+356"},{"name":"MU","code":"+230"},{"name":"MV","code":"+960"},{"name":"MW","code":"+265"},{"name":"MX","code":"+52"},{"name":"MY","code":"+60"},{"name":"MZ","code":"+258"},{"name":"NA","code":"+264"},{"name":"NC","code":"+687"},{"name":"NE","code":"+227"},{"name":"NF","code":"+672"},{"name":"NG","code":"+234"},{"name":"NI","code":"+505"},{"name":"NL","code":"+31"},{"name":"NO","code":"+47"},{"name":"NP","code":"+977"},{"name":"NR","code":"+674"},{"name":"NU","code":"+683"},{"name":"NZ","code":"+64"},{"name":"OM","code":"+968"},{"name":"PA","code":"+507"},{"name":"PE","code":"+51"},{"name":"PF","code":"+689"},{"name":"PG","code":"+675"},{"name":"PH","code":"+63"},{"name":"PK","code":"+92"},{"name":"PL","code":"+48"},{"name":"PM","code":"+508"},{"name":"PN","code":"+872"},{"name":"PR","code":"+1 939"},{"name":"PS","code":"+970"},{"name":"PT","code":"+351"},{"name":"PW","code":"+680"},{"name":"PY","code":"+595"},{"name":"QA","code":"+974"},{"name":"RE","code":"+262"},{"name":"RO","code":"+40"},{"name":"RS","code":"+381"},{"name":"RU","code":"+7"},{"name":"RW","code":"+250"},{"name":"SA","code":"+966"},{"name":"SB","code":"+677"},{"name":"SC","code":"+248"},{"name":"SD","code":"+249"},{"name":"SE","code":"+46"},{"name":"SG","code":"+65"},{"name":"SH","code":"+290"},{"name":"SI","code":"+386"},{"name":"SJ","code":"+47"},{"name":"SK","code":"+421"},{"name":"SL","code":"+232"},{"name":"SM","code":"+378"},{"name":"SN","code":"+221"},{"name":"SO","code":"+252"},{"name":"SR","code":"+597"},{"name":"ST","code":"+239"},{"name":"SV","code":"+503"},{"name":"SY","code":"+963"},{"name":"SZ","code":"+268"},{"name":"TC","code":"+1 649"},{"name":"TD","code":"+235"},{"name":"TG","code":"+228"},{"name":"TH","code":"+66"},{"name":"TJ","code":"+992"},{"name":"TK","code":"+690"},{"name":"TL","code":"+670"},{"name":"TM","code":"+993"},{"name":"TN","code":"+216"},{"name":"TO","code":"+676"},{"name":"TR","code":"+90"},{"name":"TT","code":"+1 868"},{"name":"TV","code":"+688"},{"name":"TW","code":"+886"},{"name":"TZ","code":"+255"},{"name":"UA","code":"+380"},{"name":"UG","code":"+256"},{"name":"US","code":"+1"},{"name":"UY","code":"+598"},{"name":"UZ","code":"+998"},{"name":"VA","code":"+379"},{"name":"VC","code":"+1 784"},{"name":"VE","code":"+58"},{"name":"VG","code":"+1 284"},{"name":"VI","code":"+1 340"},{"name":"VN","code":"+84"},{"name":"VU","code":"+678"},{"name":"WF","code":"+681"},{"name":"WS","code":"+685"},{"name":"YE","code":"+967"},{"name":"YT","code":"+262"},{"name":"ZA","code":"+27"},{"name":"ZM","code":"+260"},{"name":"ZW","code":"+263"}];

export const sendOTPButtonId = 'sendOTPButton';