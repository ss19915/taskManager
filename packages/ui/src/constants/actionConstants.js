export const SAVE_ALL_TASKS = 'SAVE_ALL_TASKS';