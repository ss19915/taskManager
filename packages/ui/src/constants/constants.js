export const API_STATUS = {
    INITIAL: 'INITIAL',
    SENT: 'SENT',
    RECEIVED: 'RECEIVED',
    ERROR: 'ERROR',
};