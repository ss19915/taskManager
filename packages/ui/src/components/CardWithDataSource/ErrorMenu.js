import React from 'react';
import { Status } from './constants';
import { Paper, Button } from '@react-web-app/theme';
import { ErrorMenuLabels } from './constants';
import T from 'prop-types';

class ErrorMenu extends React.PureComponent {
    static propTypes = {
        retry: T.func,
        status: T.oneOf(Object.values(Status)),
        error: T.object,
    };

    state = {
        showError: false,
    };

    toggleShowError = () => this.setState((prevState) => ({ showError: !prevState.showError }));

    render() {
        const {
            retry,
            status,
            error,
        } = this.props;

        return (
            <React.Fragment>
                <Button
                    onClick={retry}
                    disabled={status === Status.LOADING}
                >
                    {ErrorMenuLabels.RETRY}
                </Button>
                <Button onClick={this.toggleShowError}>
                    {this.state.showError ? ErrorMenuLabels.HIDE_ERROR : ErrorMenuLabels.SHOW_ERROR}
                </Button>

                {this.state.showError &&
                    <Paper css={(theme) => ({ color: theme.warningColor })}>
                        {`Error: ${JSON.stringify(error)}`}
                    </Paper>
                }
            </React.Fragment>
        );
    }
}

export default ErrorMenu;