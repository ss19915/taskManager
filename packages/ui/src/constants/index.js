import { API_STATUS } from './constants';

export {
    API_STATUS,
};