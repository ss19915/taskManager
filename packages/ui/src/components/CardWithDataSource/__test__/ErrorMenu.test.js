import React from 'react';
import ErrorMenu from '../ErrorMenu';
import { ErrorMenuLabels } from '../constants';
import { Button } from '@react-web-app/theme';

describe('Component: ErrorMenu', () => {
    it('Should render correctly with no child', () => {
        const wrapper = shallow(<ErrorMenu/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('Should call retry when clicked on Retry button', () => {
        const retry = jest.fn();

        const findRetryButton = (wrapper) => {
            let button;
            wrapper.find(Button).map( (Button) => {
                if(Button.props().children === ErrorMenuLabels.RETRY){
                    button = Button;
                }
            });

            return button;
        };
        
        const wrapper = shallow(<ErrorMenu retry={retry} />);
        
        findRetryButton(wrapper).props().onClick();
        expect(retry).toBeCalledTimes(1);
    });
});