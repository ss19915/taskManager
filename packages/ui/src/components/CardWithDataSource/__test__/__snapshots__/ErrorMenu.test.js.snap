// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorMenu Should render correctly with no child 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    disabled={false}
  >
    Retry
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    Show Error
  </WithStyles(ForwardRef(Button))>
</Fragment>
`;
